File: TeddyPet_BE/src/main/java/fpt/teddypet/application/dto/request/feedback/FeedbackRequest.java
package fpt.teddypet.application.dto.request.feedback;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.util.UUID;

public record FeedbackRequest(
        UUID token, // For guest
        UUID orderId, // For logged in user
        @NotNull(message = "Product ID is required") Long productId,
        Long variantId,
        @NotNull(message = "Rating is required") @Min(value = 1, message = "Rating must be at least 1") @Max(value = 5, message = "Rating must be at most 5") Integer rating,
        @NotBlank(message = "Comment cannot be empty") String comment) {
}

File: teddypet_mobile/lib/data/models/request/feedback/feedback_request.dart
class FeedbackRequest {
  final String? token; // For guest (though mobile usually uses orderId)
  final String? orderId; // For logged in user
  final int productId;
  final int? variantId;
  final int rating;
  final String comment;

  FeedbackRequest({
    this.token,
    this.orderId,
    required this.productId,
    this.variantId,
    required this.rating,
    required this.comment,
  });

  Map<String, dynamic> toJson() {
    return {
      if (token != null) 'token': token,
      if (orderId != null) 'orderId': orderId,
      'productId': productId,
      if (variantId != null) 'variantId': variantId,
      'rating': rating,
      'comment': comment,
    };
  }
}
